### PR TITLE
More tensor conversions from Vec and primitives

### PR DESCRIPTION
**Pull request by jerry73204**
The PR made following changes.

- Implement `From<Vec<_>> for Tensor`
- Support array or vec of bool type conversion
- More conversion tests

### CODE REVIEW

File: src/tensor/mod.rs [hunk 1]
```diff
@@ -575,15 +576,25 @@ macro_rules! try_from_impl {
                     None => failure::bail!("cannot convert to slice"),
                     Some(v) => v,
                 };
-                let tn = Self::of_slice(slice);
+                let tn = Self::f_of_slice(slice)?;
                 let shape: Vec<i64> = value.shape().iter().map(|s| *s as i64).collect();
                 Ok(tn.reshape(&shape))
```
Comment (LaurentMazare) [inline src/tensor/mod.rs]: I cannot give it a try right now but maybe it would make sense to even use `f_reshape` here?

Comment (jerry73204) [reply src/tensor/mod.rs]: Sounds good. Change it accordingly.

**Comment by LaurentMazare** [comment]
Merged, thanks for the PR!

**Comment by jerry73204** [comment]
I noticed `From<Vec<Vec<_>>>` is not implemented for `Tensor`. If so, the vector length sanity check will be done in `try_from()`. How do you think about this?

**Comment by LaurentMazare** [comment]
Sorry I just came across your last message. Would that be an issue? Failing in the try_from should return an appropriate error hopefully (but it may be good to add a test case for this).

**Comment by jerry73204** [comment]
The `Vec<Vec<_>>` type has no content of dimension size. I was concerning the case that `Tensor` size is erroneously calculated in some cases and the conversion to `Vec<Vec<_>>` will silently get the wrong size.

Maybe I worry too much. I can send a PR and implement as it is and place the dimension verification issue on other thread.

**Pull request merged**